### PR TITLE
Refactor the builder source code

### PR DESCRIPTION
**Pull request by xyproto**
Trying to avoid the long series of `if err != nil`, if possible.

### CODE REVIEW

**Comment by xyproto** [comment]
It's 5 lines shorter and introduces some constants and helper functions.

Accept or reject at will. :)

File: builder/builder.go [hunk 1]
```diff
@@ -51,223 +71,198 @@ func (b *Builder) SetMirrorList() error {
...
 
-	err = utils.SetupMakepkg(b.Container)
-	if err != nil {
+// SetupChrootConfig -
+func (b *Builder) SetupChrootConfig() error {
+	if err := b.SetupGnupg(); err != nil {
 		return err
 	}
-	return nil
+	return b.WithContainer(ContainerFunctions{
```
Comment (xyproto) [inline builder/builder.go:224]: `ContainerActions` may be a better name

Comment (Foxboron) [reply builder/builder.go:224]: Yes, thats a better name. I wonder if some of these functions can be refactored out of the builder struct completely into it's own file.

**Comment by Foxboron** [comment]
Merged and splitted the new functions into `builder/utils.go`. The plan is that some of those files should be configurable when we introduce Environment settings for the reproducible build toolset.

**Pull request merged**